src/server/game/Entities/Player/Player.cpp

```diff
@@ -2664,12 +2664,12 @@ void Player::ZoneTeleport(uint32 zoneId)
                 {
                     if (!player->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_TAXI_FLIGHT))
                     {
-                        // player->ResummonPetTemporaryUnSummonedIfAny();
                         player->SummonLastSummonedBattlePet();
                         player->RestoreAllControlled();
                     }
                     player->ProcessDelayedOperations();
                     player->ReCreateAreaTriggerObjects();
+                    player->ResummonPetTemporaryUnSummonedIfAny();
```

Comment:
Shouldn't it be in the if statement above? If the player is on a flight path I guess the pet should not be summoned?

Reply:
@ReyDonovan other than that again looks good, thanks :partying_face: